Source/ARTDefault.m

```diff
@@ -109,4 +117,37 @@ + (NSString *)bundleBuildNumber {
     return infoDictionary[ARTDefault_bundleBuildNumberKey];
 }
 
++ (NSString *)osName {
```

Comment:
In light of [my comment](https://github.com/ably/ably-cocoa/pull/1119#issuecomment-843370936) around conforming dynamic agent components (i.e. where the OS vends us a string which we have little control over), what do people think about us emitting strings with mixed case like those presented here?

I like the idea that we're going for "correctness" with these (i.e. how the vendor would represent their product in technical docs or marketing materials), however I don't know if there's any potential impact in terms of how we report on these downstream (i.e. server-side).

Also, see @owenpearson's [recent comment](https://github.com/ably/ably-cocoa/pull/1119#issuecomment-845805830).

Thoughts, @lmars, @SimonWoolf, @paddybyers, @kavalerov, @AndyNicks, Owen, @tomkirbygreen, @ben-xD?

Reply:
With respect to the characters allowed, I'd say stick with the same format as User-Agent which is what we based this off, where each product and version is a `token`, and `token` is defined here: https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6

As for mixed case, I don't see an issue with that.

Reply:
Thanks, @lmars ... any ideas why [this Stackoverflow answer](https://stackoverflow.com/a/44652895) would omit the `; any VCHAR, except delimiters` from the end of that RFC BNF? 🤔 

Just trying to imagine conformed algorithm / regular expression to validate.

Reply:
Would it help to see specific examples of the full set of data to be collected (In their worst case) to test the design/ spec, including encoding and decoding? The best (and only) example I see from [RSC7d1](https://ably-docs.herokuapp.com/client-lib-development-guide/features/#RSC7d1) is `ably-flutter/1.2.0 ably-java/1.2.1 android/24` and it now seems a bit simplistic to the case we have now?

Reply:
> any ideas why this Stackoverflow answer would omit the ; any VCHAR, except delimiters from the end of that RFC BNF?

No idea TBH 😐

Reply:
@ben-xD what I'm trying to define is the _limits_ of what we'll accept, and it looks like those limits are going to be pretty broad at the end of the day. In the scope of this PR and the Flutter example you quote from the spec, it's primarily the OS products (e.g. `macOS`) and the dynamic model products (e.g. `iPhone12,5`) that we're talking about - and, for me, that feels concrete enough.

I'll have a run at making a docs PR to reflect this.

Reply:
While this is still under discussion (https://github.com/ably/docs/pull/1096/files#r636950729), I've pushed the draft version of the possible solution. @QuintinWillison please check if I'm moving in the right direction.